### ISSUE TITLE
Include custom metadata during command dispatch

### ISSUE THREAD

**Issue by slashdotdash** [slashdotdash/commanded#61]
The supported event stores provide storing metadata per event appended to a stream. It would be useful to allow this metadata to be provided during commanded dispatch. 

An example usage would be to include the identity of the user invoking the command:

```elixir
command = %OpenAccount{account_number: "ACC123", initial_balance: 1_000}
metadata = %{"user_id" => "user-12345"}

:ok = BankRouter.dispatch(command, metadata: metdata)
```

The *optional* metadata would be recorded against any persisted events.

**Comment by sammkj** [comment]
Is anyone working on this? If not, I would like to tackle this.

**Comment by florish** [comment]
@sammkj I've had a short conversation on this on Gitter with @slashdotdash, as I was also looking for this feature. I'm not yet familiar enough with Commanded internals to help out effectively on implementing this, but I'm available for reviewing and/or testing out this new feature if needed.

**Comment by sammkj** [comment]
hey @florish, I already sent in a PR #84. Please feel free to try it out in your project! :smiley:

**Comment by florish** [comment]
@sammkj Wow, great work! I see it's already merged into master by @slashdotdash as #85. Can't wait to try it out.
